### ISSUE TITLE
OSC not working in lattest build http://casparcg.com/builds/CasparCG%20Server/master/casparcg-server-727542fc4d288c07ecebfda2e270ce3f55fc2329-windows.zip

### ISSUE THREAD

**Issue by vimlesh1975** [CasparCG/server#1087]
OSC is not working in lattest build.

http://casparcg.com/builds/CasparCG%20Server/master/casparcg-server-727542fc4d288c07ecebfda2e270ce3f55fc2329-windows.zip

**Comment by dotarmin** [comment]
@vimlesh1975, thanks for reporting this! Next time you report an issue, please consider how you write your issue. I'm referring specifically to this issue. More details about an issue means also faster help from the community and maintainers. 

> OSC is not working in lattest build.

Can you try to start playing a clip or something and check if you get any OSC messages?

Best regards,
Armin

**Comment by romanZay** [comment]
Hello guys!
I’m receiving OSC packets very well in CasparCG 2.2. But here is question. I wasn’t successful with understanding a values for /file/time address. Instead of specified in Wiki decimal values for current/total time I get a 4-byte values for both parameters. That’s not a problem, but still can’t find why value for current time isn’t start from 0? And generally values for total time is a bit strange - for 25 sec duration it is about 16844. What is this? I’m unable to use existing libraries for handling OSC, because I’m coding in LiveCode. I’ve done handler for OSC parse by myself. Almost successful, but I need to understand what means these values? Can U help me, or advise me “where to go”?
Thank you!!!

**Comment by Julusian** [comment]
@vimlesh1975 In particular, is it you are getting no osc at all, some is missing/wrong. Some STR to be able to reproduce what you are seeing would be helpful, otherwise we can't tell what is wrong.

I have just checked with the same version and am getting osc data that looks normal but didn't look into the values

**Comment by vimlesh1975** [comment]
With my casparcg client I am using bespoke library.  Version 01 August was working normal. 21 September version giving only audio  values.

**Comment by dotarmin** [comment]
> I have just checked with the same version and am getting osc data that looks normal but didn't look into the values

Same here.

**Comment by Julusian** [comment]
@vimlesh1975 could your problem just be that some of the paths have changed? https://github.com/CasparCG/server/pull/1063

**Comment by vimlesh1975** [comment]
@Julusian 
Yes, you are right. And thank you very much for you time.
Syntax has changed like
/channel/1/stage/layer/1/file/time  ---> /channel/1/stage/layer/1/foreground/file/time
/channel/1/stage/layer/1/file/path ---> /channel/1/stage/layer/1/foreground/file/path
/channel/1/stage/layer/1/file/name ---> /channel/1/stage/layer/1/foreground/file/name

**Comment by jalajgupta1604** [comment]
hello all 
i am facing problem in new casparcg version 2.2.0 there are no support for OSC and some command like cls. can any one help me. i have compiled  this version- 2018-12-29 - Version 2.2.0 Stable.

**Comment by dotarmin** [comment]
Hi @jalajgupta1604 

> in new casparcg version 2.2.0 there are no support for OSC

v2.2.0 does support OSC. 

> and some command like cls

Have you started the `scanner.exe`? It's bundled with the server but it also can be compiled on it's own: [CasparCG Media Scanner](https://github.com/CasparCG/media-scanner)

Cheers!

**Comment by vimlesh1975** [comment]
As reported [above](https://github.com/CasparCG/server/issues/1087#issuecomment-428448707) some syntax have changed. OSC works good in all version of server.

**Comment by jalajgupta1604** [comment]
so you are saying that i need to start scanner.exe. but i have compiled CasparCG latest version in visual studio 2017 and there are no any scanner.exe generated.

**Comment by dotarmin** [comment]
> so you are saying that i need to start scanner.exe. but i have compiled CasparCG latest version in visual studio 2017 and there are no any scanner.exe generated.

That's correct. It's a separate repository (see my previous post) that needs to be built. The scanner is that is bundled with the builds and not when you build the server it-self.

Cheers.

**Comment by jalajgupta1604** [comment]
@dotarmin 
ok i will do the same but there is also one question that when i downloaded server 2.2 from git not source code only compiled version by caspar only which have scanner.exe and run this scanner and caspar both but i could not be able to get OSC information. i can get information of media file by CLS command.

**Comment by hreinnbeck** [comment]
Do you have an active AMCP connection or have you predefined OSC clients in config?  There won't be any OSC output unless the same IP has an active AMCP connection or is predefined as a listener in the config.
